src/commonMain/kotlin/dev/romainguy/kotlin/math/Quaternion.kt

```diff
@@ -0,0 +1,332 @@
+    }
+
+    operator fun get(index1: Int, index2: Int, index3: Int, index4: Int): Quaternion {
+        return Quaternion(get(index1), get(index2), get(index3), get(index4))
+    }
+
+    inline operator fun invoke(index: Int) = get(index - 1)
+
+    operator fun set(index: Int, v: Float) = when (index) {
+        0 -> x = v
+        1 -> y = v
+        2 -> z = v
+        3 -> w = v
+        else -> throw IllegalArgumentException("index must be in 0..3")
+    }
+
+    operator fun set(index1: Int, index2: Int, v: Float) {
+        set(index1, v)
+        set(index2, v)
+    }
+
+    operator fun set(index1: Int, index2: Int, index3: Int, v: Float) {
+        set(index1, v)
+        set(index2, v)
+        set(index3, v)
+    }
+
+    operator fun set(index1: Int, index2: Int, index3: Int, index4: Int, v: Float) {
+        set(index1, v)
+        set(index2, v)
+        set(index3, v)
+        set(index4, v)
+    }
+
+    operator fun set(index: QuaternionComponent, v: Float) = when (index) {
+        QuaternionComponent.X -> x = v
+        QuaternionComponent.Y -> y = v
+        QuaternionComponent.Z -> z = v
+        QuaternionComponent.W -> w = v
+    }
+
+    operator fun set(index1: QuaternionComponent, index2: QuaternionComponent, v: Float) {
+        set(index1, v)
+        set(index2, v)
+    }
+
+    operator fun set(
+            index1: QuaternionComponent, index2: QuaternionComponent, index3: QuaternionComponent, v: Float) {
+        set(index1, v)
+        set(index2, v)
+        set(index3, v)
+    }
+
+    operator fun set(
+            index1: QuaternionComponent, index2: QuaternionComponent,
+            index3: QuaternionComponent, index4: QuaternionComponent, v: Float) {
+        set(index1, v)
+        set(index2, v)
+        set(index3, v)
+        set(index4, v)
+    }
+
+    operator fun unaryMinus() = Quaternion(-x, -y, -z, -w)
+
+    inline operator fun plus(v: Float) = Quaternion(x + v, y + v, z + v, w + v)
+    inline operator fun minus(v: Float) = Quaternion(x - v, y - v, z - v, w - v)
+    inline operator fun times(v: Float) = Quaternion(x * v, y * v, z * v, w * v)
+    inline operator fun div(v: Float) = Quaternion(x / v, y / v, z / v, w / v)
+
+    inline operator fun times(v: Float3) = (this * Quaternion(v, 0.0f) * inverse(this)).xyz
+
+    inline operator fun plus(q: Quaternion) = Quaternion(x + q.x, y + q.y, z + q.z, w + q.w)
+    inline operator fun minus(q: Quaternion) = Quaternion(x - q.x, y - q.y, z - q.z, w - q.w)
+    inline operator fun times(q: Quaternion) = Quaternion(
+            w * q.x + x * q.w + y * q.z - z * q.y,
+            w * q.y - x * q.z + y * q.w + z * q.x,
+            w * q.z + x * q.y - y * q.x + z * q.w,
+            w * q.w - x * q.x - y * q.y - z * q.z)
+
+    inline fun transform(block: (Float) -> Float): Quaternion {
+        x = block(x)
+        y = block(y)
+        z = block(z)
+        w = block(w)
+        return this
+    }
+
+    fun toEulerAngle() = eulerAngles(this)
```

Comment:
`toEulerAngles()`

You should also add a `toMatrix()` like there's a `toQuaternion()` on `Mat4`.

Reply:
Fixed https://github.com/romainguy/kotlin-math/pull/33/commits/e60495ba84a929cbda6ede258a067ad5baa72d13